multiversx_sdk/core/transactions_factories/token_management_transactions_factory.py

```diff
@@ -624,3 +648,216 @@ def create_transaction_for_burning_quantity(
+                                                   royalties: int) -> Transaction:
+        parts: List[str] = [
+            "ESDTModifyRoyalties",
+            arg_to_string(token_identifier),
+            arg_to_string(token_nonce),
+            arg_to_string(royalties)
+        ]
+
+        return TransactionBuilder(
+            config=self._config,
+            sender=sender,
+            receiver=sender,
+            amount=None,
+            gas_limit=self._config.gas_limit_esdt_modify_royalties,
+            add_data_movement_gas=True,
+            data_parts=parts
+        ).build()
+
+    def create_transaction_for_setting_new_uris(self,
+                                                sender: IAddress,
+                                                token_identifier: str,
+                                                token_nonce: int,
+                                                uris: List[str]) -> Transaction:
+        if not uris:
+            raise BadUsageError("No URIs provided")
+
+        parts: List[str] = [
+            "ESDTSetNewURIs",
+            arg_to_string(token_identifier),
+            arg_to_string(token_nonce),
+            *map(arg_to_string, uris)
+        ]
+
+        return TransactionBuilder(
+            config=self._config,
+            sender=sender,
+            receiver=sender,
+            amount=None,
+            gas_limit=self._config.gas_limit_set_new_uris,
+            add_data_movement_gas=True,
+            data_parts=parts
+        ).build()
+
+    def create_transaction_for_modifying_creator(self,
+                                                 sender: IAddress,
+                                                 token_identifier: str,
+                                                 token_nonce: int) -> Transaction:
+        parts: List[str] = [
+            "ESDTModifyCreator",
+            arg_to_string(token_identifier),
+            arg_to_string(token_nonce)
+        ]
+
+        return TransactionBuilder(
+            config=self._config,
+            sender=sender,
+            receiver=sender,
+            amount=None,
+            gas_limit=self._config.gas_limit_esdt_modify_creator,
+            add_data_movement_gas=True,
+            data_parts=parts
+        ).build()
+
+    def create_transaction_for_updating_metadata(self,
+                                                 sender: IAddress,
+                                                 token_identifier: str,
+                                                 token_nonce: int,
+                                                 token_name: str,
+                                                 royalties: int,
+                                                 hash: str,
+                                                 attributes: bytes,
+                                                 uris: List[str]) -> Transaction:
+        if not uris:
+            raise BadUsageError("No URIs provided")
```

Comment:
now that I think about it, this check should be deleted because someine may not want to update the `uris`.